rules/docker-gnmi-watchdog.mk

```diff
@@ -0,0 +1,33 @@
+# docker image for gnmi watchdog
+
+DOCKER_GNMI_WATCHDOG_STEM = docker-gnmi-watchdog
+DOCKER_GNMI_WATCHDOG = $(DOCKER_GNMI_WATCHDOG_STEM).gz
+DOCKER_GNMI_WATCHDOG_DBG = $(DOCKER_GNMI_WATCHDOG_STEM)-$(DBG_IMAGE_MARK).gz
+
+$(DOCKER_GNMI_WATCHDOG)_LOAD_DOCKERS = $(DOCKER_CONFIG_ENGINE_BOOKWORM)
+
+$(DOCKER_GNMI_WATCHDOG)_PATH = $(DOCKERS_PATH)/$(DOCKER_GNMI_WATCHDOG_STEM)
+
+$(DOCKER_GNMI_WATCHDOG)_VERSION = 1.0.0
+$(DOCKER_GNMI_WATCHDOG)_PACKAGE_NAME = gnmi_watchdog
+
+SONIC_DOCKER_IMAGES += $(DOCKER_GNMI_WATCHDOG)
+SONIC_BOOKWORM_DOCKERS += $(DOCKER_GNMI_WATCHDOG)
+ifeq ($(INCLUDE_GNMI), y)
+SONIC_INSTALL_DOCKER_IMAGES += $(DOCKER_GNMI_WATCHDOG)
+endif
+
+SONIC_DOCKER_DBG_IMAGES += $(DOCKER_GNMI_WATCHDOG_DBG)
+SONIC_BOOKWORM_DBG_DOCKERS += $(DOCKER_GNMI_WATCHDOG_DBG)
+ifeq ($(INCLUDE_GNMI), y)
+SONIC_INSTALL_DOCKER_DBG_IMAGES += $(DOCKER_GNMI_WATCHDOG_DBG)
+endif
+
+$(DOCKER_GNMI_WATCHDOG)_CONTAINER_NAME = gnmi_watchdog
+$(DOCKER_GNMI_WATCHDOG)_RUN_OPT += -t --privileged --pid=host
+$(DOCKER_GNMI_WATCHDOG)_RUN_OPT += -v /lib/systemd/system:/lib/systemd/system:rw
+$(DOCKER_GNMI_WATCHDOG)_RUN_OPT += -v /etc/audit:/etc/audit:rw
```

Comment:
>audit [](http://example.com/codeflow?start=43&length=5)

please double check container options. do you really need privileged? ideally use linux cap instead of privileged.

Reply:
I have removed wrong options.
I have tried --cap-add=all, nsenter command does not work in container, and we still need privileged

Reply:
Remove privileged for now